trin-core/src/jsonrpc/types.rs

```diff
@@ -70,6 +77,27 @@ fn validate_jsonrpc_version(jsonrpc: &str) -> Result<(), ValidationError> {
     Ok(())
 }
 
+pub struct PingParams {
+    pub enr: Enr,
+}
+
+impl TryFrom<Params> for PingParams {
+    type Error = ValidationError;
+
+    fn try_from(params: Params) -> Result<Self, Self::Error> {
+        match params {
+            Params::Array(val) => match val.len() {
+                1 => match Enr::from_str(val[0].as_str().unwrap()) {
```

Comment:
Similar to @carver's comment below, we would want to add another `match` on the call to [`as_str`](https://docs.serde.rs/serde_json/value/enum.Value.html#method.as_str) to check for a `None` value, which would indicate that we received some non-string value.